srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java

```diff
@@ -17,6 +33,146 @@
     
     private String modelName;
     
+    private List<Specialty> applicableSpecialties;
+    
+    private static final Logger fLogger = LoggerFactory.getLogger(EditRiskModel.class);
+
+    /**
+     * Class to hold the label, type and coefficient for a (link @ModelTerm) 
+     * and its coefficient on the Admin's Edit Model page. 
+     * 
+     * @author owner
+     *
+     */
+    public static class ModelTermSummary  implements Comparable<ModelTermSummary>
+    {
+        private final ModelTerm targetTerm;
+        
+        private String displayName;   // the name presented on the Edit Model page
+        private String termType;      // the variable type
+        
+        private ModelTermSummary( ModelTerm mt, String name, String type ) 
+        {
+            displayName = name;
+            targetTerm = mt;
+            termType = type;
+        }
+         
+        public String getDisplayName()  
+        {
+            return displayName;
+        }
+        
+        public String getTermType() 
+        {
+            return termType;
+        }
+        
+        public float getCoefficient()
+        {
+            return targetTerm.getCoefficient();
+        }
+
+        public static ModelTermSummary createTermSummary( ModelTerm modelTerm ) 
+        {
+            String dispName="";
+            String type="";
+            
+            if( modelTerm instanceof ConstantTerm ) 
```

Comment:
Instead of using an if statement, it would be much better to use a Visitor pattern here. The if/else statements are too convoluted and start to pollute the code when you get more than 3 cases. You can look into EditVarFactory, or InputParserVisitor to get a feel for the pattern. EditVarFactory is easier to read through.